include/crpropa/advectionField/AdvectionField.h

```diff
@@ -128,6 +128,112 @@ class SphericalAdvectionField: public AdvectionField {
 	std::string getDescription() const;
 };
 
+/**
+ @class OneDimensionalCartesianShock
+ @brief Advection field in x-direction with shock at x = 0 and width lShock approximated by tanh() 
+		with variable compression ratio vUp/vDown
+ */
+class OneDimensionalCartesianShock: public AdvectionField {
+	double compressionRatio; //compression ratio of shock
+	double vUp; //upstream velocity 
+	double lShock; //shock width
+public:
+/** Constructor
+	@param compressionRatio //compression ratio of shock
+	@param vUpstream //upstream velocity 
```

Comment:
This description is not the same variable name as in the constructer. This may lead to problems with the doxygen generation.